nixos/modules/services/networking/syncthing-relay.nix

```diff
@@ -0,0 +1,121 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.syncthing.relay;
+
+  dataDirectory = "/var/lib/syncthing-relay";
+
+  relayOptions = [
+    "--keys=${dataDirectory}"
+    "--listen=${cfg.listenAddress}:${toString cfg.port}"
+    "--status-srv=${cfg.statusListenAddress}:${toString cfg.statusPort}"
+    "--provided-by=${escapeShellArg cfg.providedBy}"
+    (optionalString (cfg.pools != null) "--pools=${escapeShellArg (concatStringsSep "," cfg.pools)}")
+    (optionalString (cfg.globalRateBps != null) "--global-rate=${toString cfg.globalRateBps}")
```

Comment:
I think
```nix
[
  "--keys=${dataDirectory}"
  # ...
]
++ optional (cfg.pools != null) "--pools=${escapeShellArg (concatStringsSep "," cfg.pools)}"
++ /* ... */
```

would be cleaner. This way `relayOptions` doesn't contain empty strings if the conditions are false.